internal/pkg/manifest/svc.go

```diff
@@ -73,18 +73,68 @@ func (a *Count) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 	}
 
-	if !a.Autoscaling.IsEmpty() {
+	if !c.Autoscaling.IsValid() {
+		return errUnmarshalSpot
+	}
+
+	if !c.Autoscaling.IsEmpty() {
+		// Successfully unmarshalled Autoscaling fields, return
 		return nil
 	}
 
-	if err := unmarshal(&a.Value); err != nil {
+	if err := unmarshal(&c.Value); err != nil {
 		return errUnmarshalCountOpts
 	}
 	return nil
 }
 
+// Spot is a custom type that can either be represented as a boolean, when
+// specified with a Range, or as an integer, which indicates the `desiredCount`
+// on a service that does not use application autoscaling.
+type Spot struct {
+	Enabled *bool
+	Base    *int
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the Spot
+// struct, allowing it to be unmarshalled into either an int value provided by
+// Base or a boolean value provided by Enabled.
+// This method implements the yaml.Unmarshaler (v2) interface.
+func (s *Spot) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&s.Base); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if *s.Base != 0 {
+		s.Enabled = nil
+		return nil
+	}
+
+	if err := unmarshal(&s.Enabled); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
```

Comment:
Perhaps `err` doesn't need to be handled differently here. It seems like if we get to the point of trying to unmarshal `s.Enabled`, it would mean that we've failed the attempt for `s.Base`. Hence if `unmarshal(&s.Enabled)` still returns an `yaml.TypeError`, that'd imply something is wrong with the manifest input.

Reply:
Yeah, that's what I had originally -- it was def tricky figuring out how to handle the yaml errors here since we're treating `spot` effectively as a union type -- I tried to follow the example for [BuildArgsOrString](https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/manifest/workload.go#L249) but for whatever reason the unit tests weren't behaving properly when I didn't have both error handling cases here. Part of it was also tied up in the fact that `spot` accepts an integer, but that was getting read as the value for `count` at one point, but I can try to dig further and see if I can refactor!